returnn/tf/layers/basic.py

```diff
@@ -2627,6 +2627,12 @@ def get_out_data_from_opts(cls, name, axes, sources=(), n_out=NotSpecified, out_
         # Time got merged with feature or batch.
         data.time_dim_axis = None
     data.feature_dim_axis = new_feature_dim_axis
+    # Temporarily set size_placeholder, dyn_size values will be set correctly in __init__
+    data.size_placeholder = {}
+    for old_axis, dyn_size in input_data.size_placeholder.items():
```

Comment:
Maybe make this deterministic with `sorted()`. (Although not so relevant, as `__init__` should anyway then overwrite this, but just nicer...)